### PR TITLE
Check for either ios-x86_64-simulator or ios-arm64_x86_64-simulator in module test

### PR DESCRIPTION
**Pull request by jmagman**
Prep for engine roll https://github.com/flutter/flutter/pull/85045 with recipe change https://flutter-review.googlesource.com/c/recipes/+/14780 that contains `ios-arm64_x86_64-simulator` instead of `ios-x86_64-simulator`

Passing: https://ci.chromium.org/p/flutter/builders/try/Mac%20build_ios_framework_module_test/7958?

### CODE REVIEW

File: dev/devicelab/bin/tasks/build_ios_framework_module_test.dart [hunk 1]
```diff
@@ -270,24 +270,36 @@ Future<void> _testBuildIosFramework(Directory projectDir, { bool isModule = fals
...
       'Flutter.xcframework',
-      'ios-x86_64-simulator',
+      'ios-arm64_x86_64-simulator',
       'Flutter.framework',
-      'Headers',
-      'Flutter.h',
-    ));
+    );
+
+    final bool x86SimulatorBinaryExists = exists(File(path.join(x86SimulatorFramework, 'Flutter')));
```
Comment (gw280) [inline dev/devicelab/bin/tasks/build_ios_framework_module_test.dart:290]: very minor nit: x86_64 should be shortened to x64, not x86. This looks good though.

**Review by gw280 — APPROVED** [review]

File: dev/devicelab/bin/tasks/build_ios_framework_module_test.dart (hunk 1 above)
Comment (jmagman) [reply dev/devicelab/bin/tasks/build_ios_framework_module_test.dart:290]: I'm going to delete this code within the next few days once we validate #85045 and are confident the roll won't be reverted.  Noted for future reference though.

**Comment by jmagman** [comment]
`build_ios_framework_module_test` passed, merging ASAP to unblock the engine roll.

**Comment by jmagman** [comment]
This test also successfully passed on the [engine roll](https://github.com/flutter/flutter/pull/85052) https://ci.chromium.org/p/flutter/builders/try/Mac%20build_ios_framework_module_test/7959?

**Pull request merged**